### ISSUE TITLE
EAS update not working

### ISSUE THREAD

**Issue by simontoulouze** [expo/eas-cli#1319]
### Build/Submit details page URL

https://expo.dev/accounts/actual/projects/espace-interimaire-actual/builds/86c015b9-9f47-4cdc-add6-f974e7783a2b

### Summary

When upgrading from expo 43 to expo 45 and from classic build system to new system, updates published with `eas update` doesn't update installed apps, just newly installed ones.
`expo publish` also stop working after upgrade.

### Managed or bare?

Managed

### Environment
expo SDK 45

```
$ expo doctor
┌─────────────────────────────────────────────────────────────────────────┐
│                                                                         │
│   There is a new version of expo-cli available (6.0.5).                 │
│   You are currently using expo-cli 6.0.2                                │
│   Install expo-cli globally using the package manager of your choice;   │
│   for example: `npm install -g expo-cli` to get the latest version      │
│                                                                         │
└─────────────────────────────────────────────────────────────────────────┘
✖ Dependency tree validation for expo-modules-autolinking failed. This validation is only available on Node 16+ / npm 8.
✖ Dependency tree validation for @expo/config-plugins failed. This validation is only available on Node 16+ / npm 8.
✖ Dependency tree validation for @expo/prebuild-config failed. This validation is only available on Node 16+ / npm 8.
✖ Dependency tree validation for @unimodules/core failed. This validation is only available on Node 16+ / npm 8.
✖ Dependency tree validation for @unimodules/react-native-adapter failed. This validation is only available on Node 16+ / npm 8.
✖ Dependency tree validation for react-native-unimodules failed. This validation is only available on Node 16+ / npm 8.
Some dependencies are incompatible with the installed expo package version:
 - @react-native-async-storage/async-storage - expected version: ~1.17.3 - actual version installed: 1.15.4
 - expo-constants - expected version: ~13.1.1 - actual version installed: 13.2.3
 - expo-image-manipulator - expected version: ~10.3.1 - actual version installed: 10.4.0
 - react-native-reanimated - expected version: ~2.8.0 - actual version installed: 2.6.0
Your project may not work correctly until you install the correct versions of the packages.
To install the correct versions of these packages, please run: expo doctor --fix-dependencies,
or install individual packages by running expo install [package-name ...]
```

### Reproducible demo or steps to reproduce from a blank project

Useful content in `eas.json` : 
```
{
    "ci": {
      "extends": "staging",
      "distribution": "internal",
      "channel": "ep-1612",
      "cache": {
        "key": "ep-1612"
      }
    }
}
```

Useful content in `app.json` : 
```
{
  "expo" : {
      "updates": {
        "fallbackToCacheTimeout": 0,
        "checkAutomatically": "ON_ERROR_RECOVERY",
        "url": "https://u.expo.dev/4bd1b015-ae8f-411d-a946-a40be5996235"
      },
      "runtimeVersion": {
        "policy": "sdkVersion"
      }
  }
}
```

**Comment by dsokal** [comment]
Hi, 

> updates published with eas update doesn't update installed apps,

Do you mean the apps that were using the legacy `expo publish`? `eas update` and `expo publish` are not interchangeable. They use totally separate backends so this is probably why you don't see updates from `eas update` for apps that used to use updates from `expo publish`.

> expo publish also stop working after upgrade.

What do you mean?

**Comment by simontoulouze** [comment]
Hi @dsokal,
Thanks for your response.

I misused the word "publish", my bad. In fact we had an expo sdk 43 app, and we were using expo publish and expo legacy build system. As we had to migrate to the eas build system we also decided to update expo to sdk 45 and to keep expo publish system for now.

Anyway, after setting the releaseChannel key in eas.json and setting up everything else the expo publish weren't working anymore. We manage to fix it by using the new eas update way for publishing OTA updates and now everything's good.

The only remaining "problem" is that we have to manually call the `checkForUpdateAsync` and `fetchUpdateAsync()` methods, otherwise the update is not automatically done.
